src/core/media/player_skeleton.cpp

```diff
@@ -347,9 +353,31 @@ struct media::PlayerSkeleton::Private
                 remote_error->emit(e);
             });
 
-            buffering_changed.connect([remote_buffering_changed](int value)
+            buffering_changed.connect([this, remote_buffering_changed](int value)
             {
-                remote_buffering_changed->emit(value);
+                auto now = std::chrono::steady_clock::now();
+                auto time_since_last_emission = now - buffering_signal_last_emission;
+                if (value >= 100 ||
+                    time_since_last_emission > std::chrono::milliseconds(500)) {
+                    // Send the signal immediately
+                    buffering_signal_timeout.cancel();
+                    buffering_signal_value = -1;
+                    remote_buffering_changed->emit(value);
+                    buffering_signal_last_emission = now;
+                } else {
+                    if (buffering_signal_value < 0) {
+                        buffering_signal_timeout.expires_from_now(boost::posix_time::milliseconds(500));
```

Comment:
Maybe also consider `time_since_last_emission`? If a buffering change comes 400ms after the last emission, I think I want to receive that change 100ms later, not 500ms later.

Reply:
Good idea, I'll change it!

Reply:
Done

Reply:
I did not see the change. Are you sure you pushed to the correct branch?

Reply:
It's more likely that I just forgot to push :-) Done now :-)